tss-esapi/src/context/tpm_commands/asymmetric_primitives.rs

```diff
@@ -75,8 +75,63 @@ impl Context {
         }
     }
 
-    // Missing function: ECDH_KeyGen
-    // Missing function: ECDH_ZGen
+    /// Generate an ephemeral key pair.
+    pub fn ecdh_keygen(
+        &mut self,
+        key_handle: KeyHandle,
+    ) -> Result<(TPM2B_ECC_POINT, TPM2B_ECC_POINT)> {
+        let mut z_point = null_mut();
+        let mut pub_point = null_mut();
+        let ret = unsafe {
+            Esys_ECDH_KeyGen(
+                self.mut_context(),
+                key_handle.into(),
+                self.required_session_1()?,
```

Comment:
Is this session really required?

Reply:
I'm not entirely sure how to read the TPM docs on this matter. I'll mark the session as optional for now.

Reply:
Just for completeness: no sessions are required for this op. You can tell by looking at the Command and Response diagram for `TPM2_ECDH_KeyGen` in the [Commands spec](https://trustedcomputinggroup.org/wp-content/uploads/TCG_TPM2_r1p59_Part3_Commands_pub.pdf). Quoting from the architecture part of the spec:

> The number of authorization sessions that a command will have is indicated in the command schematic in 
TPM 2.0  Part  3. If a handle in the  handle area has the "@" decoration, then an authorization session is 
required  be  present  (an  authorization  session  being  either  a  password,  a  policy  session,  or  an  HMAC 
session). 
The  authorization  sessions  occur  in  the  order  of  the  associated  entity  handles.  That  is,  the  first  handle 
with an "@" decoration will be associated with the first session in the Authorization Area. 

If you go to the same section for TPM2_ECDH_ZGen, for example, you can see that the key handle has a `@` in front, so a session is required there.

Reply:
Great, thanks for the explanation @ionut-arm! I believe now the session management should be adjusted in both cases.